### ISSUE TITLE
Reviewer comments: Kingella kingae

### ISSUE THREAD

**Issue by CFGrote** [mpievolbio-scicomp/rarefan#12]
Sophie Abby reported issues when running these IDs:

* ~92cx136~ -> 92cx136_
* b2ecb95l
* _v6qq4vm:

I had the following message on the Results page: 
“REPINs 
. There was a problem with the REP(IN) analysis output data. Please check your results carefully.” 

When is this message provided, and could it be more explicit? Is it linked to the following sentence? 

“We detected 0 REPINs in the reference genome.” 

- I got the following message on the Kingella dataset: 
“Seed sequences 
There are 0 21bp long sequences in the reference genome that occur more frequently than 55 times.” 

I don’t understand this, as there were several REPIN proposed subsequently? Including in the reference genome? Arent’ the REPIN searches based on REP found in the reference genome, as suggested by Fig. 1? Moreover, there were >70 sequences listed as overrepresented in the file “.overrep”. (example of runs “_v6qq4vm”, or run “b2ecb95l”). 

- I could not find the output file called “prox.stats” in both runs (Dodkonia and Kingella) in the downloaded folders. However, they were available in the Dodkonia “out” folder provided on Zenodo.

**Comment by CFGrote** [comment]
92cx136 is not a valid run ID, probably it should be 92cx136_

**Comment by fredeBio** [comment]
I just tried to plot the data from your link and it told me "disconnected from the server". What does this mean?

----- Original Message -----

**Comment by CFGrote** [comment]
no idea. probably another corner case that the R code does not handle. i'll investigate

**Comment by CFGrote** [comment]
Must be the data, other runs plot just fine (including example datasets)

**Comment by CFGrote** [comment]
plotREPINs() produces a tree:
![image](https://user-images.githubusercontent.com/15476807/187167662-0f795b4e-c941-4258-a4ce-d8d1ff71ddd3.png)

**Comment by CFGrote** [comment]
Das RAYT alignment file ist leer, da haengt dann der code

**Comment by fredeBio** [comment]
Ja, ich glaube das sollten wir auch fixen.

----- Original Message -----

**Comment by CFGrote** [comment]
> Ja, ich glaube das sollten wir auch fixen.

Hast Du einen Vorschlag wie?

**Comment by fredeBio** [comment]
Ich glaube es wäre cool wenn wir eine Seite mit einer Fehlermeldung zeigen würden. In diesem Fall vielleicht, "Since the dataset did not contain any REPINs, plotting will not be informative" . Oder die Fehlermeldung direkt auf der Hauptseite anzeigen und den Plotlink deaktivieren.

----- Original Message -----

**Comment by CFGrote** [comment]
Ok, das geht.

**Comment by fredeBio** [comment]
cool :)

----- Original Message -----

**Comment by CFGrote** [comment]
This is indeed confusing:
Results summary says:

# RAYTs

We discovered 1 RAYTs across all submitted genomes using tblastn with yafM_Ecoli at an e-value threshold of 1e-30.

Seed sequences

There are 0 21bp long sequences in the reference genome that occur more frequently than 55 times.

# REPINs

. There was a problem with the REP(IN) analysis output data. Please check your results carefully.

**We detected 0 REPINs in the reference genome.**


But REPIN plot show REPIN population sizes of ~170 for all strains
![image](https://user-images.githubusercontent.com/15476807/187222058-372b24dc-e883-494d-b34c-09329dc0a8a0.png)

**Comment by CFGrote** [comment]
plus, the one RAYT does not show up in the plot

**Comment by CFGrote** [comment]
Ok, so here's the problem: Rarefan (the java code) seems to truncate input sequence filenames after the first dot to generate the various output directories. In this case, the input filename GCF_900475905.1_49595_E01_genomic.fna becomes GCF_900475905 and so forth. However, the server takes the entire filenames for its dropdown menus and also for the database entries. The report generator takes filenames (e.g. for the reference strain) from the database. In the above case, it tries to read GCF_900475905.1_49595_E01_genomic_2/GCF_900475905.1_49595_E01_genomic_2_largestCluster.nodes to count the number of nmers and that fails because that file does not exist due to rarefan's file naming conventions.

I can use the same truncation scheme for the database entries if that is safe (i.e. what if different input genome filenames differ only after the first dot?). @fredeBio please comment.

**Comment by fredeBio** [comment]
Well, one solution would be to replace all dots except for the last one with an underscore. What do you think?

----- Original Message -----

**Comment by CFGrote** [comment]
yes, exactly what i'm trying to do right now

**Comment by fredeBio** [comment]
Great :)!

----- Original Message -----

**Comment by CFGrote** [comment]
that works

**Comment by CFGrote** [comment]
With same input data as above, results summary now looks like this:
![image](https://user-images.githubusercontent.com/15476807/187400838-1d6eb15e-1713-4027-861a-257f3fa95632.png)

**Comment by CFGrote** [comment]
Still confusing is why the one RAYT is not shown in the plot (REPINs tab)
![image](https://user-images.githubusercontent.com/15476807/187401107-372d8c09-80c5-484f-89dd-0290e0e58969.png)

**Comment by CFGrote** [comment]
And why the report says 260 REPINs in the reference genome while the plot shows only 3 REPINs for the reference strain

**Comment by fredeBio** [comment]
Weird! I think I will have to look at this later. so much weirdness!

**Comment by CFGrote** [comment]
So there is a RAYT but no REPINs associated to it?

**Comment by CFGrote** [comment]
>     * I could not find the output file called “prox.stats” in both runs (Dodkonia and Kingella) in the downloaded folders. However, they were available in the Dodkonia “out” folder provided on Zenodo.

That file must have been computed manually, not by rarefan.

**Comment by CFGrote** [comment]
PR #14 and #13 solve the reported inconsistencies between plots and reports. I'll close here but we should open another issue regarding phrasing of error messages and reports.
